src/libexpr/primops/fetchTree.cc

```diff
@@ -182,8 +182,26 @@ static void prim_fetchTree(EvalState & state, const PosIdx pos, Value * * args,
     fetchTree(state, pos, args, v, std::nullopt, FetchTreeParams { .allowNameArgument = false });
 }
 
-// FIXME: document
-static RegisterPrimOp primop_fetchTree("fetchTree", 1, prim_fetchTree);
+static RegisterPrimOp primop_fetchTree({
+    .name = "fetchTree",
+    .args = {"args"},
+    .arity = 1,
+    .doc = R"(
+      Introduced in 2.4
+
+      The fetcher infrastructure is exposed via the `fetchTree` built-in.
+
+      `builtins.fetchTree` allows fetching a source tree using any
+      backends supported by the fetcher infrastructure. It subsumes the
```

Comment:
Shouldn't mention "fetcher infrastructure". Instead it can refer to the docs on flake references (https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake.html#flake-references), specifically the attrset form.